src/config.js

```diff
@@ -954,24 +963,58 @@ class Config {
       const scopedSettings = newSettings
       newSettings = newSettings['*']
       delete scopedSettings['*']
-      this.resetUserScopedSettings(scopedSettings)
+      this.resetScopedSettings(scopedSettings, {source})
     }
 
     return this.transact(() => {
-      this.settings = {}
+      this._clearUnscopedSettingsForSource(source)
       this.settingsLoaded = true
-      for (let key in newSettings) { const value = newSettings[key]; this.set(key, value, {save: false}) }
+      for (let key in newSettings) {
+        const value = newSettings[key]
+        this.set(key, value, {save: false, source})
+      }
       if (this.pendingOperations.length) {
         for (let op of this.pendingOperations) { op() }
         this.pendingOperations = []
       }
     })
   }
 
+  _clearUnscopedSettingsForSource (source) {
+    switch (source) {
+      case (this.projectFile):
+        this.projectSettings = {}
+        return
+      default:
+        this.settings = {}
+    }
+  }
+
+  resetProjectSettings (newSettings, projectFile) {
+    // Sets the scope and source of all project settings to `path`.
+    newSettings = Object.assign({}, newSettings)
+    const oldProjectFile = this.projectFile
+    this.projectFile = projectFile
+    if (this.projectFile != null) {
+      this._resetSettings(newSettings, {source: this.projectFile})
+    } else {
+      this.scopedSettingsStore.removePropertiesForSource(oldProjectFile)
+      this.projectSettings = {}
+    }
+  }
+
+  clearProjectSettings () {
+    this.resetProjectSettings({}, null)
+  }
+
   getRawValue (keyPath, options = {}) {
     let value
     if (!options.excludeSources || !options.excludeSources.includes(this.mainSource)) {
       value = getValueAtKeyPath(this.settings, keyPath)
+      if (this.projectFile != null) {
+        const projectValue = getValueAtKeyPath(this.projectSettings, keyPath)
+        value = projectValue || value
```

Comment:
Are there not cases where we would want a "falsy" projectValue to trump the settings value?

Reply:
Great catch!